### ISSUE TITLE
RestRequest's RequestFormat getting reverted to Xml when calling AddBody

### ISSUE THREAD

**Issue by forbjok** [restsharp/RestSharp#1246]
After updating to 106.6.2 from 106.5.4, the code
```C#
// Construct request
var request = new RestRequest(resource, Method.POST)
{
	RequestFormat = DataFormat.Json,
};

// Add request body
request.AddBody(body);
```
no longer works.
A quick investigation shows that RequestFormat is getting reverted to Xml when AddBody is called.

In my case, I fixed this by changing the code to

```C#
// Construct request
var request = new RestRequest(resource, Method.POST);

// Add request body
request.AddJsonBody(body);
```

I don't know if this will be considered a bug or not, but it is definitely a compatibility break with previous versions, so I thought you should know about it.

## Expected Behavior
The same as before 106.6 - RequestFormat should remain set to Json after calling AddBody.

## Actual Behavior
RequestFormat automatically gets reverted to Xml after AddBody has been called, causing the request to fail.

## Steps to Reproduce the Problem

  1. Create a RestRequest with RequestFormat set to Json
  2. Call AddBody, passing an object (or null)

## Specifications

  - Version: 106.6.2
  - Platform: Windows 10 v1809, .NET Core 2.2
  - Subsystem:

## StackTrace
Probably not relevant, but included anyway.
In this case, the body object is null - which works fine with Json, but causes an exception when set to Xml.
<details>
System.NullReferenceException: Object reference not set to an instance of an object.
   at RestSharp.Serializers.XmlSerializer.Serialize(Object obj)
   at RestSharp.Serialization.Xml.XmlRestSerializer.Serialize(BodyParameter bodyParameter)
   at RestSharp.RestRequestExtensions.AddBody(IHttp http, BodyParameter bodyParameter, IDictionary`2 serializers)
   at RestSharp.RestClient.ConfigureHttp(IRestRequest request)
   at RestSharp.RestClient.ExecuteAsync(IRestRequest request, Action`2 callback, String httpMethod, Func`4 getWebRequest)
   at RestSharp.RestClient.ExecuteAsync(IRestRequest request, Action`2 callback, Method httpMethod)
   at RestSharp.RestClient.ExecuteAsync(IRestRequest request, Action`2 callback)
   at RestSharp.RestClient.ExecuteAsync[T](IRestRequest request, Action`2 callback)
   at RestSharp.RestClient.ExecuteTaskAsync[T](IRestRequest request, CancellationToken token)
--- End of stack trace from previous location where exception was thrown ---
   at MWWebToWeb.MobileWorker.Services.MwClient.ExecuteRequest[TResult](IRestRequest request) in D:\Source\bitbucket\kfkvamme\mw-integration-webtoweb\MWWebToWeb\MWWebToWeb.MobileWorker\Services\MwClient.cs:line 35
   at MWWebToWeb.MobileWorker.Services.MwClient.Get[TResult,TBody](String resource, TBody body) in D:\Source\bitbucket\kfkvamme\mw-integration-webtoweb\MWWebToWeb\MWWebToWeb.MobileWorker\Services\MwClient.cs:line 53
   at MWWebToWeb.MobileWorker.Registers.MwCustomerRegister.RetrieveUpdated(Nullable`1 since) in D:\Source\bitbucket\kfkvamme\mw-integration-webtoweb\MWWebToWeb\MWWebToWeb.MobileWorker\Registers\MwCustomerRegister.cs:line 61
   at MWWebToWeb.Common.Services.SyncRegisterService`3.SyncMultipleUp(CancellationToken cancellationToken, ISyncRegisterLock registerLock) in D:\Source\bitbucket\kfkvamme\mw-integration-webtoweb\MWWebToWeb\MWWebToWeb.Common\Services\SyncRegisterService.cs:line 169
   at MWWebToWeb.Common.Services.SyncRegisterService`3.SyncRegister(CancellationToken cancellationToken) in D:\Source\bitbucket\kfkvamme\mw-integration-webtoweb\MWWebToWeb\MWWebToWeb.Common\Services\SyncRegisterService.cs:line 118
</details>

**Comment by alexeyzimarev** [comment]
Check 106.6.3, it should be better

**Comment by alexeyzimarev** [comment]
Well, AddBody was always adding as XML by default. Please post the code that worked before.

**Comment by forbjok** [comment]
> I am wondering why you don’t use AddJsonBody?

As I said in my previous comment, I have already updated my code to use AddJsonBody.
It doesn't really matter much to me, but the current behavior breaks backwards compatibility with previous versions, so it's likely that other people than me will run into issues as well.

**Comment by alexeyzimarev** [comment]
The PR is released in 106.6.4
